core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java

```diff
@@ -498,7 +509,10 @@ protected void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) thro
                 // ignore
                 return;
             }
-            if (!NetworkExceptionHelper.isCloseConnectionException(e.getCause())) {
+            if (e.getCause() instanceof CircuitBreakingException) {
```

Comment:
I think (in general) that defining protocol level information in an exception mixes concepts and makes the exception classes more complex. I was not aware that we already do this so I agree with you that in the context of our code base we should define the status in the exception. Consequently, I have removed the special case handling for `CircuitBreakingException`.